sdk/core/azure-core-amqp/inc/azure/core/amqp/common/global_state.hpp

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <atomic>
+#include <functional>
+#include <list>
+#include <mutex>
+#include <thread>
+
+namespace Azure { namespace Core { namespace Amqp { namespace Common { namespace _detail {
+
+  /**
+   * uAMQP and azure-c-shared-util require that the platform_init and platform_uninit functions be
+   * called before using the various API functions.
+   *
+   * The GlobalState class maintains a singleton static local variable using [static local
+   * variables](https://en.cppreference.com/w/cpp/language/storage_duration#Static_local_variables),
+   * also known as "Magic Statics". This ensures that we initialize only once.
+   *
+   */
+
+  class GlobalState {
+  public:
+    GlobalState();
```

Comment:
Does constructor need to be public?
What about destructor? If the user should not be deleting the instance - it can be private too.
Copy constructor and assignment are public?

Reply:
No it doesn't need to be public.